src/ledger.ts

```diff
@@ -76,6 +78,34 @@ export class LedgerCanister {
     );
   };
 
+  public createSubaccount = async ({ name }: { name: string }) => {
+    const service = Actor.createActor<NNSDappService>(certifiedIdlFactory, {
+      agent: this.agent,
+      canisterId: this.canisterId,
+    });
+
+    const counter = 0;
+    let response = await service.create_sub_account(name);
+    while (response.AccountNotFound === null && counter < 2) {
+      // This was needed at least once, why?
+      // Otherwise I was getting a `AccountNotFound` error
+      await service.add_account();
+      response = await service.create_sub_account(name);
+    }
+
+    if (response.NameTooLong === null) {
```

Comment:
we can deconstruct the response `const {NameTooLong, SubAccountLimitExceeded, Ok} = response;`

Reply:
Yes, true.